### PR TITLE
Remove backendRoot and all compile-time dependencies on implementation subprojects

### PR DESCRIPTION
**Pull request by PawelLipski**
Basically:
* `backendRoot` got completely merged into `backendImpl`
* runtime binding machinery got extracted into a separate utility module

### CODE REVIEW

File: binding/src/main/java/com/virtuslab/binding/RuntimeBinding.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package com.virtuslab.binding;
+
+import static lombok.Lombok.sneakyThrow;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.reflections.Reflections;
+
+public final class RuntimeBinding {
```
Comment (PawelLipski) [inline binding/src/main/java/com/virtuslab/binding/RuntimeBinding.java:10]: DI for the poor xD

File: gitCoreApi/src/main/java/com/virtuslab/gitcore/api/GitCoreBranchTrackingStatus.java [hunk 2]
```diff
@@ -1,11 +1,9 @@
-package com.virtuslab.gitcore.impl.jgit;
+package com.virtuslab.gitcore.api;
 
 import lombok.Data;
 
-import com.virtuslab.gitcore.api.IGitCoreBranchTrackingStatus;
-
 @Data(staticConstructor = "of")
-public class GitCoreBranchTrackingStatus implements IGitCoreBranchTrackingStatus {
```
Comment (PawelLipski) [inline gitCoreApi/src/main/java/com/virtuslab/gitcore/api/GitCoreBranchTrackingStatus.java:8]: I don't think interface/impl split was justified for a simple and rather implementation-independent stuff like this

File: binding/src/main/java/com/virtuslab/binding/RuntimeBinding.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  private RuntimeBinding() {}
+
+  private static final Reflections reflectionsInstance = new Reflections("com.virtuslab");
+
+  /**
+   * @throws ClassNotFoundException if no implementing class found, or more than one implementing class found
+   * @throws NoSuchMethodException if the sole implementing class has no parameterless constructor
+   * @throws IllegalAccessException if the parameterless constructor of sole implementing class is not accessible
+   */
+  public static <T> T instantiateSoleImplementingClass(Class<T> interfaze) {
```
Comment (PawelLipski) [inline binding/src/main/java/com/virtuslab/binding/RuntimeBinding.java:20]: Technically, `interfaze` can point to a class, even a concrete one

File: binding/src/main/java/com/virtuslab/binding/RuntimeBinding.java (hunk 1 above)
Comment (mkondratek) [reply binding/src/main/java/com/virtuslab/binding/RuntimeBinding.java:10]: this part of the PR is the best! :rocket:

**Pull request merged**